contract/models/abstract_contract_line.py

```diff
@@ -224,12 +224,17 @@ def _inverse_price_unit(self):
         for line in self.filtered(lambda x: not x.automatic_price):
             line.specific_price = line.price_unit
 
+    def _get_price_subtotal(self):
+        self.ensure_one()
+        subtotal = self.quantity * self.price_unit
+        discount = self.discount / 100
+        subtotal *= 1 - discount
+        return subtotal
```

Comment:
A quick redesign suggestion to allow a little bit more of flexibility

```suggestion
    
    def _get_discounted_price_subtotal(self):
        """Return the discounted price subtotal"""
        subtotal = self._get_price_subtotal()
        discount = self.discount / 100
        subtotal *= 1 - discount
        return subtotal
        
    def _get_price_subtotal(self):
        """Return the subtotal price (without discount)"""
        self.ensure_one()
        return self.quantity * self.price_unit
```